src/lib/EventListener/AdminExceptionListener.php

```diff
@@ -99,6 +106,7 @@ public function onKernelException(GetResponseForExceptionEvent $event)
 
         // map exception to UI notification
         $this->notificationHandler->error($this->getNotificationMessage($exception));
+        $this->loggerInterface->critical($exception->getMessage());
```

Comment:
Add exception object to context, so it can be properly recovered by tools like Sentry or DataDog, when their loggers are added to the chain.
```suggestion
        $this->loggerInterface->critical($exception->getMessage(), [
            'exception' => $e,
        ]);
```

Additionally, I'm not 100% certain that it is a critical level error all the time. In most cases these are reserved for PHP failures, syntax issues, and similar problems when the application straight up fails to launch - and immediate administrative action is required.

```suggestion
        $this->loggerInterface->error($exception->getMessage(), [
            'exception' => $e,
        ]);
```

Reply:
I added the exception object to the context as suggested

Reply:
Regarding loglevel please see my comment in the PR description regarding merging
In principle, I agree to that log level should be error. 
However, the monolog config for 2.5 says that only messages with critical and alert levels are sent to the logs.
So if you insist, I can change log level to error but the end result will then still be that admin-ui errors will still not end up in the log in prod out-of-the-box

Reply:
@Steveb-p ^

Reply:
@vidarl I would therefore propose a different approach, one that is present in the `Symfony\Component\HttpKernel\EventListener\ErrorListener::logKernelException`. 

Instead of hard coding a log level, we can control it using any sort of logic. Even the basic property with log level injected via constructor would do - although a full blown class map as done in Symfony default listener would be neat.

And by the way, the whole reason we even need to do this logging is because our listener is on the same priority as default, aforementioned `Symfony\Component\HttpKernel\EventListener\ErrorListener::logKernelException`. If it were lower, then we could simply re-use default Symfony listener for logging and adjust the class map as needed.

See excerpt from `debug:event` with `kernel.exception`.

```shell
php bin/console debug:event kernel.exception

Registered Listeners for "kernel.exception" Event
=================================================

 ------- --------------------------------------------------------------------------------------------------------- ---------- 
  Order   Callable                                                                                                  Priority  
 ------- --------------------------------------------------------------------------------------------------------- ---------- 
  #1      BabDev\PagerfantaBundle\EventListener\ConvertNotValidMaxPerPageToNotFoundListener::onKernelException()    512       
  #2      BabDev\PagerfantaBundle\EventListener\ConvertNotValidCurrentPageToNotFoundListener::onKernelException()   512       
  #3      Ibexa\Workflow\Event\Subscriber\VersionLockedExceptionListener::onVersionLocked()                         255       
  #4      Ibexa\Personalization\Event\Listener\InvalidInstallationKeyListener::onInvalidInstallationKey()           255       
  #5      Ibexa\Bundle\Rest\EventListener\ResponseListener::onKernelExceptionView()                                 20        
  #6      Ibexa\Bundle\Core\EventListener\ExceptionListener::onKernelException()                                    10        
  #7      Ibexa\HttpCache\EventSubscriber\HiddenLocationExceptionSubscriber::tagHiddenLocationExceptionResponse()   10        
  #8      Ibexa\PageBuilder\Event\Subscriber\ExceptionSubscriber::onKernelException()                               10        
  #9      Symfony\WebpackEncoreBundle\EventListener\ExceptionListener::onKernelException()                          0         
  #10     Ibexa\AdminUi\EventListener\AdminExceptionListener::onKernelException()                                   0         
  #11     Symfony\Component\HttpKernel\EventListener\ErrorListener::logKernelException()                            0         
  #12     Symfony\Component\HttpKernel\EventListener\ProfilerListener::onKernelException()                          0         
  #13     Symfony\Bundle\SwiftmailerBundle\EventListener\EmailSenderListener::onException()                         0         
  #14     FOS\HttpCacheBundle\EventListener\InvalidationListener::onKernelException()                               0         
  #15     Ibexa\Bundle\Core\EventListener\BackgroundIndexingTerminateListener::reindex()                            0         
  #16     Symfony\Component\HttpKernel\EventListener\RouterListener::onKernelException()                            -64       
  #17     Symfony\Component\HttpKernel\EventListener\ErrorListener::onKernelException()                             -128      
 ------- --------------------------------------------------------------------------------------------------------- ----------
 ```

Reply:
@Steveb-p 
The purpose of the mapping done in `Symfony\Component\HttpKernel\EventListener\ErrorListener::logKernelException` is to have some logic so that some exceptions (decided by classname of the exception) are logged with log level X while others are logged with log level Y. I am not sure that is what we need in this use case

> Instead of hard coding a log level, we can control it using any sort of logic. Even the basic property with log level injected via constructor would do

So, I inject for instance "$logLevel" as parameter in the constructor, then we [bind](https://symfony.com/blog/new-in-symfony-3-4-local-service-binding) that value to "critical" in 2.5, but "error" in 3.x and later. Is this what you mean ?

Reply:
@Steveb-p  : still waiting for feedback on the question above.

Anyway, now that 2.5 is no longer maintained, I have rebased this to 2.3 ( Ibexa DXP 3.3 ).
As mentioned I think default log level should be error in 3.3 ( vs critical in 2.5 ).

But if you still want, we can make some logic around that as soon as you clarify my question asked in https://github.com/ezsystems/ezplatform-admin-ui/pull/2028#discussion_r864796618

Reply:
Sorry @vidarl, I've missed the original response.

> So, I inject for instance "$logLevel" as parameter in the constructor, then we [bind](https://symfony.com/blog/new-in-symfony-3-4-local-service-binding) that value to "critical" in 2.5, but "error" in 3.x and later. Is this what you mean ?

Pretty much what I'd like to see.

Developers can configure their loggin in multitude of ways. One might treat one error type as critical (and require immediate attention, including notifying the admin in the middle of the night), while another might consider this an error that can be tackled later.

Providing ability to control precisely what is logged and how should be what we strive for. For this reason I'm suggesting passing error level as constructor value (even if it's not configurable via user-friendly configuration) and tagging the service with `monolog.logger` tag with proper channel attribute (so it can be excluded from general application errors).

Reply:
@Steveb-p  : Fixed in https://github.com/ezsystems/ezplatform-admin-ui/pull/2028/commits/c2932886c24944b0fbaf196d1e15d7d51db8d557

ok ?